holmes/core/investigation_path/metrics.py

```
@@ -0,0 +1,272 @@
+
+
+def expected_calibration_error(
+    confidences: Sequence[float],
+    correct: Sequence[bool],
+    bins: int = DEFAULT_CALIBRATION_BINS,
+) -> float:
+    """Mean gap between stated confidence and observed correctness, weighted by bin size."""
+    if not confidences:
+        return 0.0
+    buckets: Dict[int, List[int]] = {}
+    for confidence, is_correct in zip(confidences, correct):
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)